keras/layers/pooling.py

```diff
@@ -424,7 +444,21 @@ class GlobalAveragePooling1D(_GlobalPooling1D):
     '''
 
     def call(self, x, mask=None):
-        return K.mean(x, axis=1)
+        if mask is None:
+            return K.mean(x, axis=1)
+        else:
+            mask = K.cast(mask, K.floatx())
+            # Expand count so we can do a broadcast division
+            count = K.sum(mask, axis=1)
+            count = K.expand_dims(count, dim=1)
+            if K.backend() == 'tensorflow':  # tensorflow shape command dosn't work
```

Comment:
We should not have this type of backend-specific branching. Why doesn't the same line work in TF? It should work, so if it doesn't then that's the problem we should solve.